runtime-modules/content-directory/src/lib.rs

```diff
@@ -929,36 +929,24 @@ decl_module! {
 
             Self::ensure_all_properties_are_valid(&new_properties)?;
 
-            let class_properties = class.get_properties_ref();
+            let class_properties = class.get_properties();
 
-            Self::ensure_all_property_names_are_unique(class_properties, &new_properties)?;
+            Self::ensure_all_property_names_are_unique(&class_properties, &new_properties)?;
 
-            // Create new Schema with existing properies provided
-            let mut schema = Schema::new(existing_properties);
+            Self::ensure_schema_properties_are_valid_indices(&existing_properties, &class_properties)?;
 
-            schema.ensure_schema_properties_are_valid_indices(class_properties)?;
+            let schema = Self::create_class_schema(existing_properties, &class_properties, &new_properties);
 
-            // Represents class properties after new `Schema` added
-            let mut updated_class_props = class.properties;
-
-            new_properties.into_iter().for_each(|prop| {
-
-                // Add new property ids to `Schema`
-                let prop_id = updated_class_props.len() as PropertyId;
-
-                schema.get_properties_mut().insert(prop_id);
-
-                // Update existing class properties
-                updated_class_props.push(prop);
-            });
+            // Update class properties after new `Schema` added
+            let updated_class_properties = Self::update_class_properties(class_properties, new_properties);
```

Comment:
can we call this `make_updated_class_properties`, function names should lead with verbs.

Reply:
Done https://github.com/iorveth/substrate-runtime-joystream/commit/3ce425c45dc66e3166be0f0f83954d3828e9bc26